src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -668,7 +676,7 @@ def mv(x):
 
         eigen_vals, eigen_vecs = e.eigenvalues, e.eigenvectors
 
-    eigen_vals = to_torch_conversion_function(eigen_vals)
-    eigen_vecs = to_torch_conversion_function(eigen_vecs)
+    eigen_vals = to_torch_conversion_function(eigen_vals).to(device)
```

Comment:
After a little digestion phase, I am not sure, if this is the correct place to add the .to(device).
The reason is the following, we have three potential computation scenarios:

1. All on CPU
2. All on GPU
3. Mixed, model forward pass on GPU -> Arnoldi iteration on CPU

Case 1 and 2 are clear, but the third case can be a valid use case, when the projections are too large to fit on GPU memory. If we force the array back on the GPU here, it could be a problem.

Maybe handling this in downstream code, i.e. solve_arnoldi, and improve documentation? What do you think?